### PR TITLE
Do not remove existing API key in scene file when no global key provided

### PR DESCRIPTION
**Pull request by rfriberg**
Fix for #353 

Note: This will _replace_ the API key in a scene file, if one has been provided. However, it will not _remove_ an existing API key, if a global key has not been provided.

Follow up:
- Move API key checks to individual components
- Consolidate API key code so it can be reused by components

### CODE REVIEW

File: src/js/components/tangram.js [hunk 1]
```diff
@@ -72,22 +102,88 @@ var TangramLayer = L.Class.extend({
...
+   *
+   * This mutates the original `config` object by necessity. Tangram does not
+   * expect it to be passed back in after it's modified.
+   *
+   * @param {Object} config - Tangram scene config object
+   * @param {string} apiKey - the API key to inject
+   */
+  _injectApiKey: function (config, apiKey) {
+
+    Object.entries(config.sources).forEach((entry) => {
```
Comment (louh) [inline src/js/components/tangram.js]: Is mapzen.js getting transpiled at all? Tangram Play has a bunch of ES2015 in it, which has some newer code that isn't supported by some browsers. I'm also not sure what browser support for mapzen.js is but thought I'd raise this flag. `Object.entries()` is a particularly [bad one](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries). `Object.assign()` (below) has [no IE support and won't work on Kelso's ancient Safari](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign).

Comment (rfriberg) [inline src/js/components/tangram.js]: Ah shoot, I forget that mapzen.js is not transpiled. Thanks for catching this!

Comment (rfriberg) [inline src/js/components/tangram.js]: Oh, interesting.  I did not run into that problem (though I seem to recall running into it in the past).  My builds seemed to run fine, aside from some super vague uglify errors.  Now assuming those are related.

Comment (louh) [inline src/js/components/tangram.js]: Yup - it seems we ran into the same thing - uglify throws an error, but the rest of the build actually completes. I misread and thought the error exited the build step!

**Comment by louh** [comment]
The UI for marking approval seems to have disappeared on me but I'm manually 👍 this, it looks good and works great!

**Comment by rfriberg** [comment]
Thanks @louh.  I'll need an approval before I can merge.  If this can wait until tomorrow, we can have @hanbyul-here take a look, too.

**Comment by louh** [comment]
I figured out how to approve!

But I am not currently in a rush as S3 is wonky and nothing is guaranteed, so it is fine to have other approvals as well.

**Comment by hanbyul-here** [comment]
For my clarity : `L.Mapzen.apiKey` overrides the key from the scene file,  but the one from the scene file doesn't. (I think we are ok with this behavior now, but maybe want to change this behavior in future since I can imagine some users being curious why mapzen.js throws api key missing warning when there is a key from scene file) 

I have tiny suggestion to put `scene.yaml` file in `test/examples/resources` folder since we already have yaml files there.

**Comment by rfriberg** [comment]
Thanks @hanbyul-here - I moved the scene file. Good eyes!  :)

What I'd like to do is move the key checks onto the individual components, so if a user really is only using vector tiles (as with Tangram Frame), then there will be no warning.  But I don't necessarily want an API key embedded in a scene file to be passed to other components, like the geocoder.  It's not a workflow I want to encourage (at least, at this point).

**Pull request merged**